### PR TITLE
feat: add more scenarios

### PR DESCRIPTION
**Pull request by michaelbeaumont**
```
• [16.749 seconds]
Simple should deploy graph
/home/mike/projects/mesh-perf/test/k8s/simple_test.go:88

  Report Entries >>
  spec.start - /home/mike/projects/mesh-perf/test/k8s/simple_test.go:67 @ 07/24/23 20:23:23.021
  duration - /home/mike/projects/mesh-perf/test/k8s/simple_test.go:123 @ 07/24/23 20:23:29.764
    6.03950113s
  spec.end - /home/mike/projects/mesh-perf/test/k8s/simple_test.go:74 @ 07/24/23 20:23:39.769
  spec.duration - /home/mike/projects/mesh-perf/test/k8s/simple_test.go:75 @ 07/24/23 20:23:39.769
    16.747684208s
  << Report Entries
------------------------------
 2023-07-24T20:23:39+02:00 retry.go:91: install yaml resource
 2023-07-24T20:23:39+02:00 logger.go:66: Running command kubectl with args [--kubeconfig /tmp/kubie-configIR9UUJ.yaml apply -f /tmp/652861821]
 2023-07-24T20:23:40+02:00 logger.go:66: meshratelimit.kuma.io/mesh-rate-limit created
• [17.276 seconds]
Simple should deploy mesh wide policy
/home/mike/projects/mesh-perf/test/k8s/simple_test.go:126

  Report Entries >>
  spec.start - /home/mike/projects/mesh-perf/test/k8s/simple_test.go:67 @ 07/24/23 20:23:39.773
  duration - /home/mike/projects/mesh-perf/test/k8s/simple_test.go:159 @ 07/24/23 20:23:47.043
    7.018134966s
  spec.end - /home/mike/projects/mesh-perf/test/k8s/simple_test.go:74 @ 07/24/23 20:23:57.045
  spec.duration - /home/mike/projects/mesh-perf/test/k8s/simple_test.go:75 @ 07/24/23 20:23:57.045
    17.272821099s
  << Report Entries
```

### CODE REVIEW

File: test/k8s/simple_test.go [hunk 1]
```diff
@@ -103,5 +118,87 @@ data:
...
+			Expect(err).ToNot(HaveOccurred())
+
+			err = cluster.Install(WaitNumPods(TestNamespace, replicas, "srv-000"))
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		It("should scale up a service", func() {
+			start := time.Now()
+			scale(2)
+			// there is no straightforward way to check if all Envoys received the config with the new endpoint, therefore we need to rely on stabilization sleep
```
Comment (lobkovilya) [inline test/k8s/simple_test.go:177]: Why do we consider "all Envoys received the config" as an assertion for the test? I think scale is successful when all instances of `srv-000` are ready

File: test/k8s/simple_test.go [hunk 2]
```diff
@@ -39,40 +43,48 @@ data:
 				Setup(cluster)
 			Expect(err).ToNot(HaveOccurred())
 			opts = append(opts,
-				WithHelmOpt("controlPlane.secrets[0].Env", "KMESH_LICENSE_INLINE"),
-				WithHelmOpt("controlPlane.secrets[0].Secret", "kong-mesh-license"),
-				WithHelmOpt("controlPlane.secrets[0].Key", "license.json"),
-			)
+				WithCtlOpts(map[string]string{
+					"--license-path": license,
```
Comment (jakubdyszkiewicz) [inline test/k8s/simple_test.go:47]: I don't think we need to manually create secret and namespace in this case. I think kumactl do it whenever we pass license-path

**Review by jakubdyszkiewicz — APPROVED** [review]

File: test/k8s/suite_test.go [hunk 3]
```diff
@@ -19,6 +20,7 @@ func TestE2E(t *testing.T) {
 }
 
 var cluster *K8sCluster
+var promClient *framework.PromClient
```
Comment (lobkovilya) [inline test/k8s/suite_test.go:23]: Is it a stateful client? If so, do we need to close something? If it's stateless, can we avoid making it global and create it when it's needed?

File: test/k8s/simple_test.go (hunk 1 above)
Comment (michaelbeaumont) [reply test/k8s/simple_test.go:161]: I think the idea is to check whether all consumers of srv-000 have the new endpoint. I'll cover this assertion in an additional PR.

File: test/k8s/simple_test.go [hunk 4]
```diff
@@ -39,56 +43,68 @@ data:
...
 	})
 
 	E2EAfterAll(func() {
 		Expect(cluster.DeleteNamespace(TestNamespace)).To(Succeed())
 		Expect(cluster.DeleteKuma()).To(Succeed())
 	})
 
+	It("should deploy the mesh", func() {
+		// just to see stabilized stats before we go further
+		Expect(true).To(BeTrue())
```
[the remark this answers is not in the document]
Comment (michaelbeaumont) [reply test/k8s/simple_test.go:68]: @jakubdyszkiewicz is this in fact necessary? Not sure what "see the stats" refers to

File: test/k8s/simple_test.go (hunk 1 above)
Comment (michaelbeaumont) [reply test/k8s/simple_test.go:165]: @lobkovilya we can discuss this in the next PR where this is actually asserted, sound good?

File: test/k8s/simple_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (jakubdyszkiewicz) [reply test/k8s/simple_test.go:68]: The idea was to also have an environment running before the test to see how much queries to db we are generating on the fresh deployment and if we are not doing anything stupid.

I'm not very attached to this, so if you think it's not a good idea, I'm ok with removing it

**Review by jakubdyszkiewicz — APPROVED** [review]

**Review by lobkovilya — APPROVED** [review]

**Pull request merged**